x-pack/plugins/search_playground/server/lib/elasticsearch_retriever.ts

```diff
@@ -82,13 +83,21 @@ export class ElasticsearchRetriever extends BaseRetriever {
 
       // default elasticsearch doc to LangChain doc
       let mapper: HitDocMapper = (hit: SearchHit<any>) => {
-        const pageContentFieldKey =
+        let pageContentFieldKey =
           typeof this.content_field === 'string'
             ? this.content_field
             : this.content_field[hit._index as string];
 
+        // field values can be a single item array or plain text, that is why needed to check the types before accessing the field
+        pageContentFieldKey = Array.isArray(pageContentFieldKey)
+          ? pageContentFieldKey[0]
+          : pageContentFieldKey;
+
+        // we need to iterate over the _source object to get the value of complex key definition such as metadata.source
+        const valueForSelectedField = getValueForSelectedField(hit._source, pageContentFieldKey);
```

Comment:
the code to transform the pageContentFieldKey into a string should be moved to route. The elasticsearch_retriever should receive content_field thats either a string or a Record<string,string>

https://github.com/elastic/kibana/blob/50d83a810744ec0d9f9ba40400e05415021482f6/x-pack/plugins/search_playground/server/routes.ts#L111